x/pairing/types/relay_exchange.go

```diff
@@ -0,0 +1,57 @@
+package types
+
+import (
+	"bytes"
+	"encoding/binary"
+	"strings"
+
+	"github.com/lavanet/lava/utils"
+)
+
+type RelayExchange struct {
+	Request RelayRequest
+	Reply   RelayReply
+}
+
+func NewRelayExchange(req RelayRequest, res RelayReply) RelayExchange {
+	return RelayExchange{Request: req, Reply: res}
+}
+
+func (re RelayExchange) GetSignature() []byte {
+	return re.Reply.Sig
+}
+
+func (re RelayExchange) PrepareForSignature() []byte {
+	re.Reply.Sig = []byte{}
+	metadataBytes := make([]byte, 0)
+	for _, metadata := range re.Reply.GetMetadata() {
+		data, err := metadata.Marshal()
+		if err != nil {
+			utils.LavaFormatError("metadata can't be marshaled to bytes", err)
+		}
+		metadataBytes = append(metadataBytes, data...)
+	}
+	// we remove the salt from the signature because it can be different
+	re.Request.RelayData.Salt = []byte{}
```

Comment:
this changes only the local copy of the `re RelayExchange`,  is that intentional?

Reply:
yes. I just need to remove it before gathering the data to sign